_posts/2019-02-14-Viewability.md

```diff
@@ -0,0 +1,115 @@
+
+##### 1. Intersection Observer
+The simplest and most reliable case are browsers which support the [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). In this case we can just call it with
+
+```javascript
+    var _ioSettings = {
+       threshold: 0.5
+    };
+    var _ioCallback = function(entry) {
+       isElementVisible = entry[0].intersectionRatio >= 0.5;
+    };
+    var elementObserver = new IntersectionObserver(_ioCallback, _ioSettings);
+    elementObserver.observe(element);
+```
+
+
+##### 2. Safari painting behavior
+When this API is not available, e.g. in older browsers, we try different methods. If the ad is running in a modern Safari browser and in an unfriendly iframe, we can make use of an optimization mechanism of Safari. It prevents the rendering of content in unfriendly iframes if they are not in the viewport. Therefore, we are putting a beacon in the middle of the iframe and check if is rendering. If it is, we know that the ad is viewable.
+
+##### 3. Scrolling
+If none of the above methods is applicable, we make use of the browser api to monitor and the scrolling of the webpage and calculate the position of the viewport ourselves. This is only possible if the adunit positioned in the topframe of the site or in a friendly iframe. We try to avoid this method, since it is relatively computing intensive in comparison to the other methods.
+
+If none of the methods is usable, we are not able to measure the viewability. In total, we are able to measure the viewability in about 80% of all delivered impressions.
+
+
+### Delivering
+
+So now we have for each impression the information if a) we could measure the viewability and b) if it was viewable.
+In general, we count unmeasurable impressions as not viewable. This is cool so far. It is very similar to a service you can buy from the IAS for instance.
+In addition, we use this information to deliver campaigns with viewable impressions as primary KPI. Thus, we want to be able to buy viewable impressions for a given price.
+The naive approach would be to take the average viewability rate of all websites and apps and decrease the bidding price by that. It is possible, that this is working, but since you would decrease the buying price by an considerable fraction, the available inventory might just be too low or too bad in quality.
+Alternatively, you could select websites and apps with a high viewability rate and only deliver on them. This approach might be feasable, but it involves a very high degree of manual optimization and you'd still need to decrease the bidding price. Plus, it only makes use of one information (the publisher) while a lot of other information might be related to the viewability rate.
+Therefore, we use machine learning as approach. The general idea is to weight each individual bid by the probability of viewability.
```

Comment:
suggest "Therefore, we use machine learning in order to weight each individual bid by the probability of viewability."